AEPAssurance/Source/AssuranceEventChunker.swift

```diff
@@ -13,28 +13,57 @@
 import AEPServices
 import Foundation
 
+/// Class that brings the capability to chunk the AssuranceEvent if in need to satisfy the socket size limit.
 struct AssuranceEventChunker {
-    let CHUNK_SIZE = 30 * 1024 // 30KB
 
+    /// The maximum size of data that an `AssuranceEvent` payload can hold after chunking
+    ///
+    /// How did we derive to 30KB?
+    ///  The maximum size of an `AssuranceEvent` to get successfully delivered through the socket is 32KB.
+    ///  AssuranceEvent consist of payload (Dictionary), type(String), vendor(String), metadata (Dictionary), timestamp(Long) and EventNumber (Integer)
+    ///  For the AssuranceEvent to completely fit into the maximum allowed socket size, we safely assign
+    ///    30KB for payload
+    ///    2KB for other fields
+    ///
+    /// Accounting for escape string bloat factor:
+    ///   After chunking the payload into multiple consumable data size, the chunk's are then recreated into corresponding `AssuranceEvent`'s.
+    /// During this process the chunked payload string is escaped and put inside the `chunkData` field of the resulting `Assurance Event`. This
+    /// escaping of string further increases the size of the chunked Data. To accommodate for unknown bloating factor, the chunk size for each event is reduced to 15KB.
+    /// Additionally the factor 0.75 corresponds to size correction due to base64 Encoding of data before sending them over Websocket.
+    let CHUNK_SIZE = (Int) ((15 * 1024) * 0.75) // 15KB
```

Comment:
if we're already shrinking the size of a chunk by half to fit inside it's 30kb limit, it's not clear to me why the extra .75 factor is still needed here. 

also, the comment of `15KB` doesn't align with the math it's describing

Reply:
The 0.75 factor is to account for the bloating due to Base64 encoding that happens right before sending the data over websocket

15KB comment is inaccurate. The actual chunk size is around 11KB

I have updated the description now.